src/Compilers/CSharp/Test/Semantic/FlowAnalysis/LocalFunctions.cs

```diff
@@ -1618,5 +1618,28 @@ void local1(string s1, out string s2)
     }
 }").VerifyDiagnostics();
         }
+
+        [Fact]
+        public void ConditionalAttributeOnLocalFunctionShouldNotProduceWarning()
+        {
+            CreateCompilation(@"
+using System.Diagnostics;
+
+public class C
+{
+    public void M()
+    {
+        local();
+        local2();
+
+        [Conditional(""DEBUG"")]
+        static void local() { }
+
+        static void local2() { }
+    }
+}
+").VerifyDiagnostics();
+        }
```

Comment:
A test where the diagnostic should be reported is already in this file:

https://github.com/dotnet/roslyn/blob/a7287667193bf2766f5b5d0b4b13f2a5c2ad8d9e/src/Compilers/CSharp/Test/Semantic/FlowAnalysis/LocalFunctions.cs#L684-L700

Feel free if you want to add more test cases.